apps/astarte_realm_management_api/test/astarte_realm_management_api_web/plugs/health_plug_test.exs

```diff
@@ -0,0 +1,93 @@
+#
+# This file is part of Astarte.
+#
+# Copyright 2025 SECO Mind Srl
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+defmodule Astarte.RealmManagement.APIWeb.HealthPlugTest do
+  use Astarte.RealmManagement.APIWeb.ConnCase, async: true
+
+  alias Astarte.RealmManagement.API.Helpers.RPCMock.DB
+  alias Astarte.RealmManagement.APIWeb.HealthPlug
+
+  describe "GET /health" do
+    test "init/1 returns nil" do
+      assert HealthPlug.init(:anything) == nil
+    end
+
+    test "returns 200 OK when status is :ready", %{conn: conn} do
+      DB.set_health_status(:READY)
+
+      conn =
+        conn
+        |> Map.put(:request_path, "/health")
+        |> Map.put(:method, "GET")
+        |> HealthPlug.call(nil)
+
+      assert conn.status == 200
+      assert conn.halted
+    end
+
+    test "returns 200 OK when status is :degraded", %{conn: conn} do
+      DB.set_health_status(:DEGRADED)
+
+      conn =
+        conn
+        |> Map.put(:request_path, "/health")
+        |> Map.put(:method, "GET")
+        |> HealthPlug.call(nil)
+
+      assert conn.status == 200
+      assert conn.halted
+    end
+
+    test "returns 503 when status is :bad", %{conn: conn} do
+      DB.set_health_status(:BAD)
+
+      conn =
+        conn
+        |> Map.put(:request_path, "/health")
+        |> Map.put(:method, "GET")
+        |> HealthPlug.call(nil)
+
+      assert conn.status == 503
+      assert conn.halted
+    end
+
+    test "returns 503 when DB returns invalid status", %{conn: conn} do
+      DB.set_health_status(:SOMETHING_ELSE)
```

Comment:
In this case, the only value you can receive is `:ERROR` (see [here](https://github.com/astarte-platform/astarte/blob/d5ed3c0a6c1af3b620a436ae506462c756537f6d/apps/astarte_realm_management/lib/astarte_realm_management/rpc/handler.ex#L74)). I would suggest using that value